### ISSUE TITLE
pubsub: metrics from pubsub server

### ISSUE THREAD

**Issue by nassiharel** [googleapis/google-cloud-go#6830]
**Is your feature request related to a problem? Please describe.**

I’m doing a scale up based on GCP PubSub metrics, I’m able to scale successfully.
But… there is a big delay until the metrics are available in GCP monitoring, ~2min.

feature: add function that returns the num of undelivered messages in real-time.

I believe that if it was possible it would be inside this package, but maybe I’m wrong.

From GCP:
subscription/num_undelivered_messages
Number of unacknowledged messages (a.k.a. backlog messages) in a subscription. 
Sampled every 60 seconds. After sampling, data is not visible for up to 120 seconds.

https://cloud.google.com/monitoring/api/metrics_gcp#pubsub/subscription/num_undelivered_messages

**Describe the solution you'd like**
A clear and concise description of what you want to happen. What should the API be? How would you use it?

Up to you :)

**Comment by hongalex** [comment]
Yeah unfortunately the Pub/Sub library is specifically for accessing the admin operations and data plane operations. For metrics, that would have to be exposed through our monitoring APIs.

**Comment by hongalex** [comment]
Closing since this isn't something that we're looking to implement in our client library but rather exposed through [Monitoring](https://cloud.google.com/pubsub/docs/monitoring).
